Update view

naributler opened the pull request.

naributler committed:
Add update item view

This view will allow the user to update the title of a
file they have uploaded to AWS.

naributler committed:
Changed value of submit button

Changed from "Upload" to "Update"

naributler committed:
Build out PATCH/UPDATE function

Added onUpdateItem, which will be activated the the "Update" button
is clicked. This will take in form data, and update the data object.

Added updateItem to API calls, which submits the AJAX PATCH request.

Tested using a curl script, and it was successful. Since the back-end
is not yet storing the updated title, we can't see the change on
the front-end.

naributler committed:
Started building out update handlebar feature

Added a data-id to each instance of an update button
Created a placeholder function, called onSelectUpdate, which is
attached to the handlebars glyphicon-pencil button.
Used jQuery to retrieve and console log that id

Next steps are to build out a SHOW function so that we can bring
back data about a single upload and populate the update view.

jim-moody commented on assets/scripts/uploads/events.js:
```diff
@@ -14,14 +14,31 @@ const onShowAllUploads = function () {
+}
+
+const onSelectUpdate = function () {
+  event.preventDefault()
```
Probably don't need this since it's not a submit action

naributler commented:
Handlebars feature needs more work - closing this pull request and will reopen when the feature is built out.

naributler committed:
Commented out token

naributler committed:
Adds feature to update title of uploaded file

When we click the pencil icon next to a row, that will populate
the "update a file" form will data about the object. Populating
the form requires a SHOW request.

Then, a PATCH request will run when the user clicks the "update"
button. This will update the title.

jim-moody commented on assets/scripts/templates/update-item-view.handlebars:
```diff
@@ -0,0 +1,13 @@
+<h2>Update Item View</h2>
+<div>
+  <form id="update-item" enctype="multipart/form-data">
```
enctype probably not needed here?
